SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -784,6 +784,483 @@ void CFlowOutput::Set_CpInverseDesign(CSolver *solver, const CGeometry *geometry
 
 }
 
+void CFlowOutput::Add_NearfieldInverseDesignOutput(){
+
+  AddHistoryOutput("EQUIVALENT_AREA",   "CEquiv_Area",  ScreenOutputFormat::SCIENTIFIC, "EQUIVALENT_AREA", "Equivalent area", HistoryFieldType::COEFFICIENT);
+  
+}
+
+void CFlowOutput::Set_NearfieldInverseDesign(CSolver *solver, const CGeometry *geometry, const CConfig *config){
+
+  ofstream EquivArea_file, FuncGrad_file;
+  unsigned short iMarker = 0;
+  su2double auxXCoord, auxYCoord, auxZCoord, InverseDesign = 0.0, DeltaX,
+    Coord_i, Coord_j, jp1Coord, *Coord = nullptr, MeanFunction,
+    *Face_Normal = nullptr, auxArea, auxPress, jFunction, jp1Function;
+  unsigned long jVertex, iVertex, iPoint, nVertex_NearField = 0, auxPoint,
+    auxDomain;
+  unsigned short iPhiAngle;
+  ofstream NearFieldEA_file; ifstream TargetEA_file;
+
+  su2double XCoordBegin_OF = config->GetEA_IntLimit(0);
+  su2double XCoordEnd_OF = config->GetEA_IntLimit(1);
+
+  unsigned short nDim = geometry->GetnDim();
+  su2double AoA = -(config->GetAoA()*PI_NUMBER/180.0);
+  su2double EAScaleFactor = config->GetEA_ScaleFactor(); // The EA Obj. Func. should be ~ force based Obj. Func.
+
+  const su2double Mach  = config->GetMach();
+  const su2double Gamma = config->GetGamma();
+  const su2double Beta = sqrt(Mach*Mach-1.0);
```

Comment:
👍 on the lower ones. While you're at it, you might try to get others (like the upper ones) const as well if possible. Now that there is a working testcase